### PR TITLE
[FIX-#6505][Dao] upgrade the MySQL driver package for building MySQL jdbcUrl

### LINKED ISSUE #6505: [Feature][Dao] Upgrade com.mysql.jdbc.Driver
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Since we have upgraded the mysql connector to 8.0.15 #6484, it's necessary to use `com.mysql.cj.jdbc.Driver` the `com.mysql.jdbc.Driver` has been deprecated.

Also note if we need to modify the current connection parameters, e.g. `useSSL=false`

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yangzi1104**
## Purpose of the pull request

fix #6505 upgrade the MySQL driver package for building MySQL jdbcUrl

## Brief change log

* replace com.mysql.jdbc.Driver everywhere with com.mysql.cj.jdbc.Driver
* update related unit tests

### CODE REVIEW

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/datasource/mysql/MysqlDatasourceProcessor.java [hunk 1]
```diff
@@ -51,7 +51,7 @@
 
     private static final String ALLOW_URL_IN_LOCAL_IN_FILE_NAME = "allowUrlInLocalInfile";
 
-    private static final String APPEND_PARAMS = "allowLoadLocalInfile=false&autoDeserialize=false&allowLocalInfile=false&allowUrlInLocalInfile=false";
+    private static final String APPEND_PARAMS = "allowLoadLocalInfile=false&autoDeserialize=false&allowLocalInfile=false&allowUrlInLocalInfile=false&useSSL=false";
```
Comment (ruanwenjun) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/datasource/mysql/MysqlDatasourceProcessor.java]: I am not sure set `useSSL=false` here is reasonable.

Comment (yangzi1104) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/datasource/mysql/MysqlDatasourceProcessor.java]: Okay, let me try without adding it for now

Comment (ruanwenjun) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/datasource/mysql/MysqlDatasourceProcessor.java]: Please rebase the upstream code and resolve the conflicts.

Comment (yangzi1104) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/datasource/mysql/MysqlDatasourceProcessor.java]: already solved

**Review by zhongjiajie — DISMISSED** [review]
LGTM now.

**Comment by zhongjiajie** [comment]
Hi @lyggyhmm , to target specific issue from PR, we recommend use keyword `fix: #issue_id` or `close: #issue_id` or `closes: #issue_id` in your PR describe. I would not only connect issue to PR but also close issue automatically when PR is be closed.

**Comment by zhongjiajie** [comment]
Wait a minute, why should we add file `dolphinscheduler-dao/src/main/resources/datasource.properties` here? @lyggyhmm 
does it add by accident

**Comment by lyggyhmm** [comment]
> 等一下，我们为什么要`dolphinscheduler-dao/src/main/resources/datasource.properties`在这里添加文件？@lyggyhmm 是不是偶然添加的

**Comment by yangzi1104** [comment]
> Wait a minute, why should we add file `dolphinscheduler-dao/src/main/resources/datasource.properties` here? @lyggyhmm does it add by accident

It seems that there was this file before. It is not enough to see that it was deleted on the 14th in the submission record. It may be that I had a little accident during fetch.

**Comment by yangzi1104** [comment]
> Hi @lyggyhmm , to target specific issue from PR, we recommend use keyword `fix: #issue_id` or `close: #issue_id` or `closes: #issue_id` in your PR describe. I would not only connect issue to PR but also close issue automatically when PR is be closed.



> Hi @lyggyhmm , to target specific issue from PR, we recommend use keyword `fix: #issue_id` or `close: #issue_id` or `closes: #issue_id` in your PR describe. I would not only connect issue to PR but also close issue automatically when PR is be closed.

Thank you for your guidance. This is the first time I mention PR here. I may not do well in some places. I will continue to improve in the future.

**Comment by zhongjiajie** [comment]
> > Hi @lyggyhmm , to target specific issue from PR, we recommend use keyword `fix: #issue_id` or `close: #issue_id` or `closes: #issue_id` in your PR describe. I would not only connect issue to PR but also close issue automatically when PR is be closed.
> 
> > Hi @lyggyhmm , to target specific issue from PR, we recommend use keyword `fix: #issue_id` or `close: #issue_id` or `closes: #issue_id` in your PR describe. I would not only connect issue to PR but also close issue automatically when PR is be closed.
> 
> Thank you for your guidance. This is the first time I mention PR here. I may not do well in some places. I will continue to improve in the future.

Hi @lyggyhmm you could reopen this PR, and just edited PR describe, we would help you finish your work.
BTW, code look good to me, you just need to remove unnecessary `dolphinscheduler-dao/src/main/resources/datasource.properties`

**Comment by yangzi1104** [comment]
> > > Hi @lyggyhmm , to target specific issue from PR, we recommend use keyword `fix: #issue_id` or `close: #issue_id` or `closes: #issue_id` in your PR describe. I would not only connect issue to PR but also close issue automatically when PR is be closed.
> > 
> > 
> > > Hi @lyggyhmm , to target specific issue from PR, we recommend use keyword `fix: #issue_id` or `close: #issue_id` or `closes: #issue_id` in your PR describe. I would not only connect issue to PR but also close issue automatically when PR is be closed.
> > 
> > 
> > Thank you for your guidance. This is the first time I mention PR here. I may not do well in some places. I will continue to improve in the future.
> 
> Hi @lyggyhmm you could reopen this PR, and just edited PR describe, we would help you finish your work. BTW, code look good to me, you just need to remove unnecessary `dolphinscheduler-dao/src/main/resources/datasource.properties`

Now, I modified the PR according to your prompts and re-edited the PR description. Please review

**Review by zhongjiajie — APPROVED** [review]
LGTM, @ruanwenjun could you please take a look? If all thing done maybe it could merge to dev now

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/datasource/mysql/MysqlDatasourceProcessor.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/datasource/mysql/MysqlDatasourceProcessor.java]: @lyggyhmm Everything looks good, one minor may need to change.
Do we require adding  `useSSL=false` in `datasource.properties` ? I worried that some new user may meet problem here.

Comment (lyggyhmm) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/datasource/mysql/MysqlDatasourceProcessor.java]: The MySQL 8.X version does not seem to throw any warnings on the useSSL configuration, which has been tested on my own server. Of course, my knowledge is limited, I want to know what insights you have

Comment (yangzi1104) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/datasource/mysql/MysqlDatasourceProcessor.java]: @ruanwenjun hi,the MySQL 8.X version does not seem to throw any warnings on the useSSL configuration, which has been tested on my own server. Of course, my knowledge is limited, I want to know what insights you have

Comment (yangzi1104) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/datasource/mysql/MysqlDatasourceProcessor.java]: Of course, configuring useSSL=false should be more secure, after all, we do not use SSL certification

Comment (yangzi1104) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/datasource/mysql/MysqlDatasourceProcessor.java]: I have removed the useSSL=false configuration last week

Comment (ruanwenjun) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/datasource/mysql/MysqlDatasourceProcessor.java]: That's ok.

**Review by ruanwenjun — APPROVED** [review]
LGTM, If no other concern, I will merge this pr tomorrow.

**Comment by ruanwenjun** [comment]
@lyggyhmm Pr merged, thanks for your contribution, this is your first contribution, you can contact to @zhongjiajie or me by email, we will help you join developer discussion groups.

**Comment by zhongjiajie** [comment]
Hi @lyggyhmm , thanks for your contribution and welcome to join community 🎉 . If you want contribute but could not find issue, maybe you could start in #5689 or just search our issue list https://github.com/apache/dolphinscheduler/issues. Looking forward to your next contribution

**Comment by yangzi1104** [comment]
> @lyggyhmm Pr merged, thanks for your contribution, this is your first contribution, you can contact to @zhongjiajie or me by email, we will help you join developer discussion groups.

Thank you for your affirmation! In my spare time, I will continue to pay attention to the dynamics of the DS community and continue to contribute my own strength! I have been in the DS seed incubator group before, and would like to ask if I can join the contributor group now

**Comment by yangzi1104** [comment]
> Hi @lyggyhmm , thanks for your contribution and welcome to join community 🎉 . If you want contribute but could not find issue, maybe you could start in #5689 or just search our issue list https://github.com/apache/dolphinscheduler/issues. Looking forward to your next contribution

Thank you for your affirmation! It’s an honor to join the community, and I still need your advice in the future

**Comment by zhongjiajie** [comment]
> Thank you for your affirmation! In my spare time, I will continue to pay attention to the dynamics of the DS community and continue to contribute my own strength! I have been in the DS seed incubator group before, and would like to ask if I can join the contributor group now

@lyggyhmm Will connect you in seed incubator wechat group

**Pull request merged**